packages/datadog-plugin-http2/src/client.js

```diff
@@ -62,7 +62,9 @@ class Http2ClientPlugin extends ClientPlugin {
 
     addHeaderTags(span, headers, HTTP_REQUEST_HEADERS, this.config)
 
-    this.tracer.inject(span, HTTP_HEADERS, headers)
+    if (!hasAmazonSignature(headers, path)) {
```

Comment:
Why does http2 work differently than http1?

Reply:
this commit of mine from Dec 1, 2023 added a config option to enable tracing headers for aws sdk but only to http requests not http2:
https://github.com/DataDog/dd-trace-js/pull/3836
at that point the decision logic diverged between http / http2
also this commit from 2021 adds the propagationFilter function to http but not to http2 (assuming http2 existed at that point, but at any rate the function was never copied to http2):
https://github.com/DataDog/dd-trace-js/pull/1602